File: 724-find-pivot-index/724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    if (nums.length === 0) return -1;
    if (nums.length === 1) return 0;
    
    let leftSum = 0;
    let totalSum = 0;
    
    for (var j = 0; j < nums.length; j++) {
        totalSum += nums[j]
    }
    
    for (var i = 0; i < nums.length; i++) {
        if (leftSum === totalSum - leftSum - nums[i]) {
            return i;
        }
        leftSum += nums[i]
    }
    return -1;
};

File: 724-find-pivot-index/724-find-pivot-index.cs
public class Solution {
    public int PivotIndex(int[] nums) {
        if (nums.Length == 0 ) return -1;
        if (nums.Length == 1 ) return 0;
        
        int leftSum = 0;
        int totalSum = 0;
        
        foreach (int num in nums)
        {
            totalSum += num;
        }
        
        for (int i = 0; i < nums.Length; i++)
        {
            if (leftSum == totalSum - leftSum - nums[i])
            {
                //returns the INDEX number -> i 
                return i;
            }
            leftSum +=  nums[i];
        }
        return -1;
    }
}

